src/account/accountService.js

```diff
@@ -0,0 +1,34 @@
+import _ from 'lodash'
+
+import { getToken } from 'account/auth'
+import { TERRASO_API_URL } from 'config'
+
+const getURL = provider => fetch(
+  new URL(`/auth/${provider}/authorize`, TERRASO_API_URL).href,
+  { headers: { 'Content-Type': 'application/json' } }
+)
+  .then(response => response.json())
+  .then(response => response.request_url)
+
+export const getAuthURLs = () => Promise.all([
+  getURL('google'),
+  getURL('apple')
+])
+  .then(([google, apple]) => ({ google, apple }))
+
+export const fetchUser = () => fetch(
+  new URL('/auth/user', TERRASO_API_URL).href,
+  {
+    headers: {
+      Authorization: `Bearer ${getToken()}`,
+      'Content-Type': 'application/json'
+    }
+  }
+)
+  .then(response => {
+    if (_.includes([401, 403], response.status)) {
+      return Promise.reject('UNAUTHENTICATED')
```

Comment:
401 is unauthorized (no creds); 403 is forbidden (creds but not access to this resource).

Reply:
For me in this particular case (getting the logged user data) 403 and 401 mean that something is wrong with your "session" that is why here I consider both to force the relogin